### PR TITLE
hashing: - done;

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -64,8 +64,15 @@
...
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-war-plugin</artifactId>
+                <version>3.2.3</version>
+                <configuration>
+                    <webXml>src\main\webapp\web.xml</webXml>
+                </configuration>
+            </plugin>
```
Comment (KseniiaMakarova) [inline pom.xml]: are you using this?

File: src/main/java/internet/shop/model/User.java [hunk 2]
```diff
@@ -22,13 +23,31 @@ public User(String login, String password) {
...
 
     public User(Long id, String name, String login, String password) {
-        this(login, password, name);
         this.id = id;
+        this.name = name;
+        this.login = login;
+        this.password = password;
+    }
+
+    public User(String name, String login, String password, byte[] salt) {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/model/User.java]: wow... isn't it too many constructors there? let's have 2, maximum 3
and set the rest of the fields

or use pattern Builder

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -15,8 +16,9 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDb = userService.findByLogin(login);
-
-        if (userFromDb.isPresent() && userFromDb.get().getPassword().equals(password)) {
+        byte[] salt = userFromDb.get().getSalt();
+        String saltedPass = HashUtil.hashPassword(password, salt);
+        if (saltedPass.equals(userFromDb.get().getPassword())) {
```
Comment (KseniiaMakarova) [inline src/main/java/internet/shop/security/AuthenticationServiceImpl.java]: why are you no more checking if the value is present?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
also,

1. in your InjectDataController you save admin to DB first, and then set roles for him. Is it ok?
2. in `init_db.sql` you are not creating `salt` column for user

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java (hunk 3 above)
Comment (jah701) [reply src/main/java/internet/shop/security/AuthenticationServiceImpl.java]: Idea suggested me to remove it as its always whether  true or false

File: pom.xml (hunk 1 above)
Comment (jah701) [reply pom.xml]: Yes? Isn't it for our ".war's" ?  Can't load project without it, mvn checkstyle fails.

File: src/main/java/internet/shop/model/User.java (hunk 2 above)
Comment (jah701) [reply src/main/java/internet/shop/model/User.java:37]: Оно не подтягивается гитом, не знаю что с ним делать..
image

**Pull request merged**